core/src/main/c/share/net.c

```diff
@@ -363,7 +406,61 @@ JNIEXPORT jint JNICALL Java_io_questdb_network_Net_getPeerPort
 JNIEXPORT jint JNICALL Java_io_questdb_network_Net_connectAddrInfo
         (JNIEnv *e, jclass cl, jint fd, jlong lpAddrInfo) {
     struct addrinfo *addr = (struct addrinfo *) lpAddrInfo;
-    return connect((int) fd, addr->ai_addr, (int) addr->ai_addrlen);
+    int result;
+
+    result = connect((int) fd, addr->ai_addr, (int) addr->ai_addrlen);
+
+    if (result == -1 && errno == EINTR) {
```

Comment:
lets extract a function (code duplication is here). Also may be try using epoll instead of select

Reply:
done